WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/EUShippingNoticeTopBannerFactory.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+final class EUShippingNoticeTopBannerFactory {
+    static func createTopBanner(onDismissPressed: @escaping () -> Void,
+                                onLearnMorePressed: @escaping (URL?) -> Void) -> TopBannerView {
+        let learnMoreAction = TopBannerViewModel.ActionButton(title: Localization.learnMore) { _ in
+            onLearnMorePressed(URL(string: String.shippingCustomsInstructionsForEUCountries))
+        }
+
+        let dismissAction = TopBannerViewModel.ActionButton(title: Localization.dismiss) { _ in
+            onDismissPressed()
+        }
+
+        let viewModel = TopBannerViewModel(title: nil,
+                                           infoText: Localization.info,
+                                           icon: .infoOutlineImage,
+                                           iconTintColor: .accent,
+                                           isExpanded: true,
+                                           topButton: .none,
+                                           actionButtons: [learnMoreAction, dismissAction])
+        let topBannerView = TopBannerView(viewModel: viewModel)
+        topBannerView.translatesAutoresizingMaskIntoConstraints = false
+        return topBannerView
+    }
+}
+
+private extension EUShippingNoticeTopBannerFactory {
+    enum Localization {
+        static let info = NSLocalizedString("When shipping to countries that follow European Union (EU) customs rules, " +
+                                            "you must provide a clear, specific description of every item. Otherwise, " +
+                                            "shipments may be delayed or interrupted at customs.",
+                                            comment: "The EU notice banner content describing why some countries require special customs description")
+        static let learnMore = NSLocalizedString("Learn more", comment: "Label for the banner Learn more button")
+        static let dismiss = NSLocalizedString("Dismiss", comment: "Label for the banner Dismiss button")
+    }
+
+    enum String {
+        static let shippingCustomsInstructionsForEUCountries = "https://www.usps.com/international/new-eu-customs-rules.htm"
+    }
```

Comment:
Oh, I see! I considered adding it to the `WooConstants` because of the safe `asURL` declaration. Still, I got confused as I noticed that all URLs there are Automattic related and decided to do something like what I saw in the `FancyAlertViewController+UnifiedLogin`, but keeping a unified source of URLs is much better indeed!

Fixed in https://github.com/woocommerce/woocommerce-ios/pull/9636/commits/e0c8ab06f1972fac0dccb4361db4a269ae5a4537 and https://github.com/woocommerce/woocommerce-ios/pull/9636/commits/9a70383a5ee39febb5d61e50574048b5f563e7a5

Thanks!

Reply:
> I noticed that all URLs there are Automattic related

That's true, but my understanding is that "trusted" in this case is "trusted to be well-formed" (rather than "internal to Automattic) because when we call `asURL()` the app will crash if the URL is malformed. This is in contrast to non-constants that we're getting on the fly and can't trust to be well-formed URLs.